pkg/resource/db_cluster/custom_update.go

```diff
@@ -608,16 +608,18 @@ func (rm *resourceManager) newCustomUpdateRequestPayload(
 		}
 		res.SetVpcSecurityGroupIds(f23)
 	}
-        // For ServerlessV2ScalingConfiguration, MaxCapacity and MinCapacity,  both need appear in modify call to be modified
+	// For ServerlessV2ScalingConfiguration, MaxCapacity and MinCapacity,  both need appear in modify call to be modified
 	if r.ko.Spec.ServerlessV2ScalingConfiguration != nil && delta.DifferentAt("Spec.ServerlessV2ScalingConfiguration") {
 		f23 := &svcsdk.ServerlessV2ScalingConfiguration{}
-                if r.ko.Spec.ServerlessV2ScalingConfiguration.MaxCapacity != nil && r.ko.Spec.ServerlessV2ScalingConfiguration.MinCapacity != nil {
-                        if delta.DifferentAt("Spec.ServerlessV2ScalingConfiguration.MaxCapacity") || delta.DifferentAt("Spec.ServerlessV2ScalingConfiguration.MinCapacity") {
-                                f23.SetMaxCapacity(*r.ko.Spec.ServerlessV2ScalingConfiguration.MaxCapacity)
-                                f23.SetMinCapacity(*r.ko.Spec.ServerlessV2ScalingConfiguration.MinCapacity)
-                        }
-                }
-                res.SetServerlessV2ScalingConfiguration(f23)
+		if r.ko.Spec.ServerlessV2ScalingConfiguration.MaxCapacity != nil && delta.DifferentAt("Spec.ServerlessV2ScalingConfiguration.MaxCapacity") {
+			f23.SetMaxCapacity(*r.ko.Spec.ServerlessV2ScalingConfiguration.MaxCapacity)
+			f23.SetMinCapacity(*r.ko.Spec.ServerlessV2ScalingConfiguration.MinCapacity)
+		}
+		if r.ko.Spec.ServerlessV2ScalingConfiguration.MinCapacity != nil && delta.DifferentAt("Spec.ServerlessV2ScalingConfiguration.MinCapacity") {
+			f23.SetMaxCapacity(*r.ko.Spec.ServerlessV2ScalingConfiguration.MaxCapacity)
```

Comment:
This will cause a panic if `r.ko.Spec.ServerlessV2ScalingConfiguration == nil`. Please remove lines 616 and 619

Reply:
Because the new api `ServerlessV2ScalingConfiguration` need both MinCapacity and MaxCapacity to appear, so whenever min or max get changed, we have to add both param to `f23`. 

I'm adding a nil check for both min and max in next revision and in either case, we set both min/max value at same time. 
With some more thinking, I think it may not be able to avoid the problem that silently fail when one val is set as nil.